grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy

```diff
@@ -213,16 +213,24 @@ class InvoiceService {
     def updateItems(Invoice invoice, List items) {
         items.each { item ->
             InvoiceItem invoiceItem = InvoiceItem.get(item.id)
+            // update existing invoice item
             if (invoiceItem) {
-                invoiceItem.quantity = item.quantity
+                if (item.quantity > 0) {
+                    invoiceItem.quantity = item.quantity
+                } else {
+                    removeInvoiceItem(invoiceItem.id)
+                }
             } else {
-                InvoiceItemCandidate candidateItem = InvoiceItemCandidate.get(item.id)
-                if (!candidateItem) {
-                    throw new IllegalArgumentException("No Invoice Item Candidate found with ID ${item.id}")
+                // create new invoice item from candidate
+                if (item.quantityToInvoice > 0) {
+                    InvoiceItemCandidate candidateItem = InvoiceItemCandidate.get(item.id)
+                    if (!candidateItem) {
+                        throw new IllegalArgumentException("No Invoice Item Candidate found with ID ${item.id}")
```

Comment:
I am trying to find a better way how to avoid nesting those conditions...
but nothing better comes to my mind, the only thing that can be done is just return from ifs, but in the case of 3 levels of nesting it will be probably unreadable. Maybe other developers will have some ideas on how to make it look cleaner.

Reply:
yes, I had the same problem with this.
I had a version that would return in the first if, basically simulating the `continue` in the for loop, but in the end thought that maybe if/else will be better, because some developers might assume that this return breaks out of the for each loop instead of skipping the iteration.